### PR TITLE
Feature/postgresql support

### PR DESCRIPTION
**Pull request by daniel-almeida-konkconsulting**

### CODE REVIEW

File: Backbone.API/Configuration/ChallengesConfiguration.cs [hunk 1]
```diff
@@ -21,5 +21,9 @@ public class SqlDatabase
 {
     [Required]
     [MinLength(1)]
-    public string ConnectionString { get; set; } = "";
+    public string Provider { get; set; } = string.Empty;
```
Comment (tnotheis) [inline Backbone.API/Configuration/ChallengesConfiguration.cs:24]: I just had an idea. We should use the `RegularExpression` attribute to allow only valid providers. 

```
[RegularExpression("SqlServer|Postgres")]
```

Can you do the same for `EventBusConfiguration#Vendor` and all `BlobStorageConfiguration#CloudProvider` please?

File: Backbone.API/appsettings.override.json [hunk 2]
```diff
@@ -30,6 +30,8 @@
       "Application": {},
       "Infrastructure": {
         "SqlDatabase": {
+          "Provider": "SqlServer",
+          //"ConnectionString": "User ID=challenges;Password=Passw0rd;Server=localhost;Port=5432;Database=enmeshed;"
```
Comment (tnotheis) [inline Backbone.API/appsettings.override.json:34]: Forgot to remove this?

File: Modules/Challenges/src/Challenges.Infrastructure/Persistence/IServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -8,23 +7,41 @@ namespace Backbone.Modules.Challenges.Infrastructure.Persistence;
...
+    private const string POSTGRES_MIGRATIONS_ASSEMBLY = "Challenges.Infrastructure.Database.Postgres";
+
     public static void AddDatabase(this IServiceCollection services, Action<DbOptions> setupOptions)
     {
         var options = new DbOptions();
         setupOptions?.Invoke(options);
 
-        services.AddDbContext<ApplicationDbContext>(dbContextOptions =>
-            dbContextOptions.UseSqlServer(options.DbConnectionString, sqlOptions =>
+        services.AddDbContext<ChallengesDbContext>(dbContextOptions => _ = options.Provider switch
```
Comment (tnotheis) [inline Modules/Challenges/src/Challenges.Infrastructure/Persistence/IServiceCollectionExtensions.cs:20]: Wouldn't the following be more readable?
```cs
switch(options.Provider) {
    case SQLSERVER: 
        services.AddDbContext<ChallengesDbContext>(dbContextOptions => 
            dbContextOptions.UseSqlServer(options.DbConnectionString, sqlOptions =>
            {
                sqlOptions.CommandTimeout(20);
                sqlOptions.MigrationsAssembly(SQLSERVER_MIGRATIONS_ASSEMBLY);
                sqlOptions.EnableRetryOnFailure(options.RetryOptions.MaxRetryCount, TimeSpan.FromSeconds(options.RetryOptions.MaxRetryDelayInSeconds), null);
            })
        );
        break;
    case SQLSERVER: 
        services.AddDbContext<ChallengesDbContext>(dbContextOptions => 
            dbContextOptions.UseNpgsql(options.DbConnectionString, sqlOptions =>
            {
                sqlOptions.CommandTimeout(20);
                sqlOptions.MigrationsAssembly(POSTGRES_MIGRATIONS_ASSEMBLY);
                sqlOptions.EnableRetryOnFailure(options.RetryOptions.MaxRetryCount, TimeSpan.FromSeconds(options.RetryOptions.MaxRetryDelayInSeconds), null);
            }),
        );
        break;
}
```

(probably not valid C#, but you get the idea. ;-)

File: Modules/Relationships/src/Relationships.Infrastructure.Database.Postgres/Relationships.Infrastructure.Database.Postgres.csproj [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+	<BaseOutputPath>..\Relationships.Infrastructure\bin\</BaseOutputPath>
```
Comment (tnotheis) [inline Modules/Relationships/src/Relationships.Infrastructure.Database.Postgres/Relationships.Infrastructure.Database.Postgres.csproj:7]: This line doesn't exist in the other projects. So there are 2 possibilities:
1. You didn't run the application with a Postgres database
2. The BaseOutputPath is not necessary after all.

I hope it's the option 2. :-)

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Relationships/src/Relationships.Infrastructure.Database.Postgres/Relationships.Infrastructure.Database.Postgres.csproj (hunk 4 above)
Comment (daniel-almeida-konkconsulting) [reply Modules/Relationships/src/Relationships.Infrastructure.Database.Postgres/Relationships.Infrastructure.Database.Postgres.csproj:7]: It's option 2!

File: docker-compose/docker-compose.yml [hunk 5]
```diff
@@ -109,6 +121,7 @@ volumes:
   rabbitmq-volume:
   mssql-server-volume:
     external: true
+  postgres-volume:
```
Comment (tnotheis) [inline docker-compose/docker-compose.yml:124]: Can you make it external, please?

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: docker-compose/docker-compose.yml [hunk 6]
```diff
@@ -58,6 +58,27 @@ services:
...
+  postgres:
+    container_name: bkb-postgres
+    hostname: postgres
+    image: postgres
+    environment:
+      - POSTGRES_PASSWORD=admin
+    volumes:
+      - postgres-volume:/var/lib/postgresql/data
+    ports:
+      - 5432:5432
```
Comment (tnotheis) [inline docker-compose/docker-compose.yml:70]: Comment it out

File: docker-compose/docker-compose.yml [hunk 7]
```diff
@@ -113,6 +134,8 @@ volumes:
   rabbitmq-volume:
   mssql-server-volume:
     external: true
+  postgres-volume:
```
Comment (tnotheis) [inline docker-compose/docker-compose.yml:137]: Comment it out

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**